src/libOpenImageIO/color_ocio.cpp

```diff
@@ -49,6 +51,64 @@ namespace OCIO = OCIO_NAMESPACE;
 OIIO_NAMESPACE_BEGIN
 
 
+// Class used as the key to index color processors in the cache.
+class ColorProcCacheKey {
+public:
+    ColorProcCacheKey (ustring in, ustring out,
+                       ustring key=ustring(), ustring val=ustring(),
+                       ustring looks=ustring(), ustring display=ustring(),
+                       ustring view=ustring(), ustring file=ustring(),
+                       bool inverse=false)
+        : inputColorSpace(in), outputColorSpace(out),
+          context_key(key), context_value(val), looks(looks), file(file),
+          inverse(inverse)
+    {
+        hash = (inputColorSpace.hash() ^ outputColorSpace.hash() ^
+                context_key.hash() ^ context_value.hash() ^
+                looks.hash() ^ display.hash() ^ view.hash() ^ file.hash())
+             + (inverse?6421:0);
+    }
+
+    friend bool operator< (const ColorProcCacheKey& a, const ColorProcCacheKey& b) {
+        if (a.hash < b.hash) return true;
+        if (b.hash < a.hash) return false;
+        // They hash the same, so now compare for real
+        if (a.inputColorSpace < b.inputColorSpace) return true;
+        else if (b.inputColorSpace < a.inputColorSpace) return false;
```

Comment:
`else` is not needed here and make the conditions a bit harder to read.

Ideally the string comparisons would be only done once instead of twice each too.

Reply:
Oh, I used the wrong operator. Changing all these to foo.c_str() < bar.c_str() because there's no reason it needs to be alphabetical.